gamemode/server/rounds.lua

```diff
@@ -113,55 +120,89 @@ local function StartRound()
     round.current = round.current + 1
     round.startTime = CurTime()
     round.state = ROUND_IN
-    GAMEMODE.HuntersReleased = false
+    round.huntersReleased = false
+
+    print("Round " .. round.current .. " is starting")
+
+    for _, v in pairs(player.GetAll()) do
+        -- remove god mode from everyone
+        v:GodDisable()
+
+        -- reset previous round data
+        v:SetPropAngleLocked(false)
+        v:SetPropLockedAngle(Angle(0,0,0))
+        v:SetPropPitchEnabled(false)
+        v:SetPropAngleSnapped(false)
+        v:SetPropRollAngle(0)
+        -- taunt data
+        v:SetLastTauntTime(CurTime())
+        v:SetLastTauntDuration(1)
+
+        if (v:Team() == TEAM_HUNTERS) then
+            -- freeze all the hunters
+            v:Freeze(true)
+            -- Give ammo to hunters scaled with number of props on map
+            v:GiveAmmo(math.floor(GetGlobalInt("NumPropsOnMap", 200) / 35 + 0.5), "AR2AltFire", true)
+
+            v:SetPlayerColor(PlayerToAccentColor(v))
+        end
+    end
+
     hook.Call("OBJHUNT_RoundStart")
+
+    -- send data to clients
+    SendRoundUpdate(function() return net.Broadcast() end)
 end
 
 local function InRound()
     local roundTime = CurTime() - round.startTime
-    -- make sure we have not gone over time
-    if (roundTime >= OBJHUNT_ROUND_TIME) then
-        round.state = ROUND_END
-        round.endTime = CurTime()
-        round.winner = "Props"
-        hook.Call("OBJHUNT_RoundEnd")
-        return
-    end
-
-    -- make sure there is at least one living player left per team
     local hunters = GetLivingPlayers(TEAM_HUNTERS)
     local props = GetLivingPlayers(TEAM_PROPS)
 
-    if (#props == 0) then
-        round.state = ROUND_END
-        round.endTime = CurTime()
-        round.winner = "Hunters"
-        hook.Call("OBJHUNT_RoundEnd")
-        return
+    -- Has anyone won yet?
+    local winner = nil
+    if (roundTime >= OBJHUNT_ROUND_TIME or #hunters == 0) then
```

Comment:
I know this is merged already but do we know that this is going to work if the last hunter kills the last prop and themselves in the same tick. I feel like that should still count as a hunter win. I think it did before.